core/src/test/java/org/apache/iceberg/TestScansAndSchemaEvolution.java

```diff
@@ -113,4 +116,80 @@ public void testPartitionSourceRename() throws IOException {
 
     assertThat(tasks).hasSize(1);
   }
+
+  @Test
+  public void testAddColumnWithDefaultValueAndQuery() throws IOException {
+    // Only enable the test for versions above V3 since default values require V3+.
+    for (int version : V3_AND_ABOVE) {
```

Comment:
I recommend that instead of iterating through the format versions, you can convert this test to a parametrised case, like the same on in this file. Using assumptions, you can skip format version below 3, something like this:
```
  @TestTemplate
  public void testAddColumnWithDefaultValueAndQuery() throws IOException {
    Assumptions.assumeTrue(V3_AND_ABOVE.contains(formatVersion), "Only enable the test for versions above V3 since default values require V3+.");
    File location = Files.createTempDirectory(temp, "junit").toFile();
```

This would then nicely show in the final report, if a certain format version above 3 breaks the column default values. What do you think?

Reply:
@nandorKollar That sounds like a good idea. Done.